### ISSUE TITLE
Different counts in ES & Pg

### ISSUE THREAD

**Issue by mdoering** [Sp2000/colplus-backend#145]
When checking the ES index for dataset 1027 and comparing it to the [pg metrics](http://api.col.plus/dataset/1027/import) there are only `7976` docs in elastic, but many more in postgres. Maybe an async indexing issue?
```
  "verbatimCount": 110325,
  "nameCount": 60834,
  "taxonCount": 45238,

``` 

``` 
// 20180927224724
// http://db.col.plus:9200/nu1027/_stats

{
  "_shards": {
    "total": 1,
    "successful": 1,
    "failed": 0
  },
  "_all": {
    "primaries": {
      "docs": {
        "count": 7976,
        "deleted": 0
      },
      "store": {
        "size_in_bytes": 3134900
      },
      "indexing": {
        "index_total": 7976,
        "index_time_in_millis": 1015,
        "index_current": 0,
        "index_failed": 0,
        "delete_total": 0,
        "delete_time_in_millis": 0,
        "delete_current": 0,
        "noop_update_total": 0,
        "is_throttled": false,
        "throttle_time_in_millis": 0
      },
      "get": {
        "total": 0,
        "time_in_millis": 0,
        "exists_total": 0,
        "exists_time_in_millis": 0,
        "missing_total": 0,
        "missing_time_in_millis": 0,
        "current": 0
      },
      "search": {
        "open_contexts": 0,
        "query_total": 0,
        "query_time_in_millis": 0,
        "query_current": 0,
        "fetch_total": 0,
        "fetch_time_in_millis": 0,
        "fetch_current": 0,
        "scroll_total": 0,
        "scroll_time_in_millis": 0,
        "scroll_current": 0,
        "suggest_total": 0,
        "suggest_time_in_millis": 0,
        "suggest_current": 0
      },
      "merges": {
        "current": 0,
        "current_docs": 0,
        "current_size_in_bytes": 0,
        "total": 0,
        "total_time_in_millis": 0,
        "total_docs": 0,
        "total_size_in_bytes": 0,
        "total_stopped_time_in_millis": 0,
        "total_throttled_time_in_millis": 0,
        "total_auto_throttle_in_bytes": 20971520
      },
      "refresh": {
        "total": 6,
        "total_time_in_millis": 178,
        "listeners": 0
      },
      "flush": {
        "total": 1,
        "periodic": 0,
        "total_time_in_millis": 103
      },
      "warmer": {
        "current": 0,
        "total": 3,
        "total_time_in_millis": 1
      },
      "query_cache": {
        "memory_size_in_bytes": 0,
        "total_count": 0,
        "hit_count": 0,
        "miss_count": 0,
        "cache_size": 0,
        "cache_count": 0,
        "evictions": 0
      },
      "fielddata": {
        "memory_size_in_bytes": 0,
        "evictions": 0
      },
      "completion": {
        "size_in_bytes": 0
      },
      "segments": {
        "count": 5,
        "memory_in_bytes": 81678,
        "terms_memory_in_bytes": 68129,
        "stored_fields_memory_in_bytes": 1984,
        "term_vectors_memory_in_bytes": 0,
        "norms_memory_in_bytes": 6656,
        "points_memory_in_bytes": 113,
        "doc_values_memory_in_bytes": 4796,
        "index_writer_memory_in_bytes": 0,
        "version_map_memory_in_bytes": 0,
        "fixed_bit_set_memory_in_bytes": 0,
        "max_unsafe_auto_id_timestamp": -1,
        "file_sizes": {
          
        }
      },
      "translog": {
        "operations": 7976,
        "size_in_bytes": 3438569,
        "uncommitted_operations": 0,
        "uncommitted_size_in_bytes": 55,
        "earliest_last_modified_age": 0
      },
      "request_cache": {
        "memory_size_in_bytes": 0,
        "evictions": 0,
        "hit_count": 0,
        "miss_count": 0
      },
      "recovery": {
        "current_as_source": 0,
        "current_as_target": 0,
        "throttle_time_in_millis": 0
      }
    },
    "total": {
      "docs": {
        "count": 7976,
        "deleted": 0
      },
      "store": {
        "size_in_bytes": 3134900
      },
      "indexing": {
        "index_total": 7976,
        "index_time_in_millis": 1015,
        "index_current": 0,
        "index_failed": 0,
        "delete_total": 0,
        "delete_time_in_millis": 0,
        "delete_current": 0,
        "noop_update_total": 0,
        "is_throttled": false,
        "throttle_time_in_millis": 0
      },
      "get": {
        "total": 0,
        "time_in_millis": 0,
        "exists_total": 0,
        "exists_time_in_millis": 0,
        "missing_total": 0,
        "missing_time_in_millis": 0,
        "current": 0
      },
      "search": {
        "open_contexts": 0,
        "query_total": 0,
        "query_time_in_millis": 0,
        "query_current": 0,
        "fetch_total": 0,
        "fetch_time_in_millis": 0,
        "fetch_current": 0,
        "scroll_total": 0,
        "scroll_time_in_millis": 0,
        "scroll_current": 0,
        "suggest_total": 0,
        "suggest_time_in_millis": 0,
        "suggest_current": 0
      },
      "merges": {
        "current": 0,
        "current_docs": 0,
        "current_size_in_bytes": 0,
        "total": 0,
        "total_time_in_millis": 0,
        "total_docs": 0,
        "total_size_in_bytes": 0,
        "total_stopped_time_in_millis": 0,
        "total_throttled_time_in_millis": 0,
        "total_auto_throttle_in_bytes": 20971520
      },
      "refresh": {
        "total": 6,
        "total_time_in_millis": 178,
        "listeners": 0
      },
      "flush": {
        "total": 1,
        "periodic": 0,
        "total_time_in_millis": 103
      },
      "warmer": {
        "current": 0,
        "total": 3,
        "total_time_in_millis": 1
      },
      "query_cache": {
        "memory_size_in_bytes": 0,
        "total_count": 0,
        "hit_count": 0,
        "miss_count": 0,
        "cache_size": 0,
        "cache_count": 0,
        "evictions": 0
      },
      "fielddata": {
        "memory_size_in_bytes": 0,
        "evictions": 0
      },
      "completion": {
        "size_in_bytes": 0
      },
      "segments": {
        "count": 5,
        "memory_in_bytes": 81678,
        "terms_memory_in_bytes": 68129,
        "stored_fields_memory_in_bytes": 1984,
        "term_vectors_memory_in_bytes": 0,
        "norms_memory_in_bytes": 6656,
        "points_memory_in_bytes": 113,
        "doc_values_memory_in_bytes": 4796,
        "index_writer_memory_in_bytes": 0,
        "version_map_memory_in_bytes": 0,
        "fixed_bit_set_memory_in_bytes": 0,
        "max_unsafe_auto_id_timestamp": -1,
        "file_sizes": {
          
        }
      },
      "translog": {
        "operations": 7976,
        "size_in_bytes": 3438569,
        "uncommitted_operations": 0,
        "uncommitted_size_in_bytes": 55,
        "earliest_last_modified_age": 0
      },
      "request_cache": {
        "memory_size_in_bytes": 0,
        "evictions": 0,
        "hit_count": 0,
        "miss_count": 0
      },
      "recovery": {
        "current_as_source": 0,
        "current_as_target": 0,
        "throttle_time_in_millis": 0
      }
    }
  },
  "indices": {
    "nu1027": {
      "uuid": "ouqNMQAGSWmU6TbgnOjrAw",
      "primaries": {
        "docs": {
          "count": 7976,
          "deleted": 0
        },
        "store": {
          "size_in_bytes": 3134900
        },
        "indexing": {
          "index_total": 7976,
          "index_time_in_millis": 1015,
          "index_current": 0,
          "index_failed": 0,
          "delete_total": 0,
          "delete_time_in_millis": 0,
          "delete_current": 0,
          "noop_update_total": 0,
          "is_throttled": false,
          "throttle_time_in_millis": 0
        },
        "get": {
          "total": 0,
          "time_in_millis": 0,
          "exists_total": 0,
          "exists_time_in_millis": 0,
          "missing_total": 0,
          "missing_time_in_millis": 0,
          "current": 0
        },
        "search": {
          "open_contexts": 0,
          "query_total": 0,
          "query_time_in_millis": 0,
          "query_current": 0,
          "fetch_total": 0,
          "fetch_time_in_millis": 0,
          "fetch_current": 0,
          "scroll_total": 0,
          "scroll_time_in_millis": 0,
          "scroll_current": 0,
          "suggest_total": 0,
          "suggest_time_in_millis": 0,
          "suggest_current": 0
        },
        "merges": {
          "current": 0,
          "current_docs": 0,
          "current_size_in_bytes": 0,
          "total": 0,
          "total_time_in_millis": 0,
          "total_docs": 0,
          "total_size_in_bytes": 0,
          "total_stopped_time_in_millis": 0,
          "total_throttled_time_in_millis": 0,
          "total_auto_throttle_in_bytes": 20971520
        },
        "refresh": {
          "total": 6,
          "total_time_in_millis": 178,
          "listeners": 0
        },
        "flush": {
          "total": 1,
          "periodic": 0,
          "total_time_in_millis": 103
        },
        "warmer": {
          "current": 0,
          "total": 3,
          "total_time_in_millis": 1
        },
        "query_cache": {
          "memory_size_in_bytes": 0,
          "total_count": 0,
          "hit_count": 0,
          "miss_count": 0,
          "cache_size": 0,
          "cache_count": 0,
          "evictions": 0
        },
        "fielddata": {
          "memory_size_in_bytes": 0,
          "evictions": 0
        },
        "completion": {
          "size_in_bytes": 0
        },
        "segments": {
          "count": 5,
          "memory_in_bytes": 81678,
          "terms_memory_in_bytes": 68129,
          "stored_fields_memory_in_bytes": 1984,
          "term_vectors_memory_in_bytes": 0,
          "norms_memory_in_bytes": 6656,
          "points_memory_in_bytes": 113,
          "doc_values_memory_in_bytes": 4796,
          "index_writer_memory_in_bytes": 0,
          "version_map_memory_in_bytes": 0,
          "fixed_bit_set_memory_in_bytes": 0,
          "max_unsafe_auto_id_timestamp": -1,
          "file_sizes": {
            
          }
        },
        "translog": {
          "operations": 7976,
          "size_in_bytes": 3438569,
          "uncommitted_operations": 0,
          "uncommitted_size_in_bytes": 55,
          "earliest_last_modified_age": 0
        },
        "request_cache": {
          "memory_size_in_bytes": 0,
          "evictions": 0,
          "hit_count": 0,
          "miss_count": 0
        },
        "recovery": {
          "current_as_source": 0,
          "current_as_target": 0,
          "throttle_time_in_millis": 0
        }
      },
      "total": {
        "docs": {
          "count": 7976,
          "deleted": 0
        },
        "store": {
          "size_in_bytes": 3134900
        },
        "indexing": {
          "index_total": 7976,
          "index_time_in_millis": 1015,
          "index_current": 0,
          "index_failed": 0,
          "delete_total": 0,
          "delete_time_in_millis": 0,
          "delete_current": 0,
          "noop_update_total": 0,
          "is_throttled": false,
          "throttle_time_in_millis": 0
        },
        "get": {
          "total": 0,
          "time_in_millis": 0,
          "exists_total": 0,
          "exists_time_in_millis": 0,
          "missing_total": 0,
          "missing_time_in_millis": 0,
          "current": 0
        },
        "search": {
          "open_contexts": 0,
          "query_total": 0,
          "query_time_in_millis": 0,
          "query_current": 0,
          "fetch_total": 0,
          "fetch_time_in_millis": 0,
          "fetch_current": 0,
          "scroll_total": 0,
          "scroll_time_in_millis": 0,
          "scroll_current": 0,
          "suggest_total": 0,
          "suggest_time_in_millis": 0,
          "suggest_current": 0
        },
        "merges": {
          "current": 0,
          "current_docs": 0,
          "current_size_in_bytes": 0,
          "total": 0,
          "total_time_in_millis": 0,
          "total_docs": 0,
          "total_size_in_bytes": 0,
          "total_stopped_time_in_millis": 0,
          "total_throttled_time_in_millis": 0,
          "total_auto_throttle_in_bytes": 20971520
        },
        "refresh": {
          "total": 6,
          "total_time_in_millis": 178,
          "listeners": 0
        },
        "flush": {
          "total": 1,
          "periodic": 0,
          "total_time_in_millis": 103
        },
        "warmer": {
          "current": 0,
          "total": 3,
          "total_time_in_millis": 1
        },
        "query_cache": {
          "memory_size_in_bytes": 0,
          "total_count": 0,
          "hit_count": 0,
          "miss_count": 0,
          "cache_size": 0,
          "cache_count": 0,
          "evictions": 0
        },
        "fielddata": {
          "memory_size_in_bytes": 0,
          "evictions": 0
        },
        "completion": {
          "size_in_bytes": 0
        },
        "segments": {
          "count": 5,
          "memory_in_bytes": 81678,
          "terms_memory_in_bytes": 68129,
          "stored_fields_memory_in_bytes": 1984,
          "term_vectors_memory_in_bytes": 0,
          "norms_memory_in_bytes": 6656,
          "points_memory_in_bytes": 113,
          "doc_values_memory_in_bytes": 4796,
          "index_writer_memory_in_bytes": 0,
          "version_map_memory_in_bytes": 0,
          "fixed_bit_set_memory_in_bytes": 0,
          "max_unsafe_auto_id_timestamp": -1,
          "file_sizes": {
            
          }
        },
        "translog": {
          "operations": 7976,
          "size_in_bytes": 3438569,
          "uncommitted_operations": 0,
          "uncommitted_size_in_bytes": 55,
          "earliest_last_modified_age": 0
        },
        "request_cache": {
          "memory_size_in_bytes": 0,
          "evictions": 0,
          "hit_count": 0,
          "miss_count": 0
        },
        "recovery": {
          "current_as_source": 0,
          "current_as_target": 0,
          "throttle_time_in_millis": 0
        }
      }
    }
  }
}
```

**Comment by ayco-at-naturalis** [comment]
This issue is likely related to #201

**Comment by ayco-at-naturalis** [comment]
The NameUsageMapper.xml did not wrap the issues within an association (of javaType VerbatimRecord). This caused MyBatis to skip all records without issues. Fixed this. The consequence is though that NameUsageWrapper (an API object) now has a verbatimRecord fields which contains (just) the issues. I have experimented with an extra, intermediate class, just for the MyBatis output, which resembles, but now quite is a NameUsageWrapper. But you would then have 3 closely related classes

1. NameUsageWrapper (The API class)
2. EsNameUsage (The ES document structure)
3. The intermediate class

It would probably actually have been (very) slightly more performant, but the code got surprisingly messy and contrived, so ditched it. Can still re-introduce it if we really like the NameUsageWrapper to stay as it was.
